Library/Homebrew/cmd/uses.rb

```diff
@@ -110,15 +108,9 @@ def self.intersection_of_dependents(use_runtime_dependents, used_formulae, args:
       deps
     else
       all = args.eval_all?
-      if args.all?
-        unless all
-          odisabled "brew uses --all",
-                    "brew uses --eval-all or HOMEBREW_EVAL_ALL"
-        end
-        all = true
-      end
 
       if !args.installed? && !(all || Homebrew::EnvConfig.eval_all?)
+        # This odisabled should probably stick around indefinitely.
         odisabled "brew uses", "brew uses --eval-all or HOMEBREW_EVAL_ALL"
```

Comment:
These should probably become a `UsageError` rather than disabled, since it isn't “disabled” when `HOMEBREW_EVAL_ALL` is set.